specification/containerinstance/resource-manager/Microsoft.ContainerInstance/stable/2018-06-01/containerInstance.json

```diff
@@ -0,0 +1,1355 @@
+        "x-ms-examples": {
+          "ContainerGroupsDelete": {
+            "$ref": "./examples/ContainerGroupsDelete.json"
+          }
+        },
+        "summary": "Delete the specified container group.",
+        "description": "Delete the specified container group in the specified subscription and resource group. The operation does not delete other resources provided by the user, such as volumes.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ContainerGroupNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/ContainerGroup"
+            }
+          },
+          "204": {
+            "description": "No Content - the specified container group was not found."
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.ContainerInstance/operations": {
+      "get": {
+        "tags": [
+          "Operations"
+        ],
+        "operationId": "Operations_List",
+        "x-ms-examples": {
+          "OperationsList": {
+            "$ref": "./examples/OperationsList.json"
+          }
+        },
+        "description": "List the operations for Azure Container Instance service.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/OperationListResult"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.ContainerInstance/locations/{location}/usages": {
+      "get": {
+        "operationId": "ContainerGroupUsage_List",
+        "x-ms-examples": {
+          "ContainerUsage": {
+            "$ref": "./examples/ContainerGroupUsage.json"
+          }
+        },
+        "description": "Get the usage for a subscription",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/LocationParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/UsageListResult"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContainerInstance/containerGroups/{containerGroupName}/containers/{containerName}/logs": {
+      "get": {
+        "operationId": "ContainerLogs_List",
```

Comment:
There are some semantics that we introduce to OperationIDs in AutoRest. Chiefly, we use the prefix here to denote an "operation group". For some languages, each operation group gets is own Client type.

For that reason, I'd recommend having fewer Operation Groups:

`ContainerGroupUsage_List` -> `ContainerGroups_ListUsage`
`ContainerLogs_List` -> `Container_ListLogs`
`StartContainer_LaunchExec` -> `Container_StartExecLaunch`

Reply:
I took another look and found another thing we may want to tweak, not necessary to fix it though! Just acknowledge if you don't want to fix it. :)

Reply:
@marstr I would rather save those for the next PR so we can get this June release out. I can update those changes quickly for another PR.

Reply:
It'd be a breaking change in the future, so we should do it now or never. Your choice though. :)